### PR TITLE
Replace HTTP Party by Rest Client

### PR DESCRIPTION
**Pull request by ignaciojonas**

### CODE REVIEW

File: lib/auth0/mixins/httpproxy.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      %i(get post post_file put patch delete).each do |method|
+        define_method(method) do |path, body = {}|
+          safe_path = URI.escape(path)
+          body = body.delete_if { |_, v| v.nil? }
+          result = if [:get, :delete].include?(method)
+                      call(method, url(safe_path), timeout, add_headers({params: body}))
+                   elsif method == :post_file
+                      call(:post, url(safe_path), timeout, headers, body)
+                   else
+                      call(method, url(safe_path), timeout, headers, body.to_json)
```
Comment (ignaciojonas) [inline lib/auth0/mixins/httpproxy.rb:18]: The difference between this line and line 16 is just the .to_json. What happen if we use the body without to_json here?

Comment (leosoubeste) [inline lib/auth0/mixins/httpproxy.rb:18]: I get an error on boolean values if body is not json.

``` Auth0::BadRequest:
       {"error":"bad.email_verified","error_description":"error in email_verified - invalid type: string (expected boolean)"}
```

Do you know how to handle that?

File: lib/auth0/api/v2/resource_servers.rb [hunk 2]
```diff
@@ -11,7 +11,7 @@ module ResourceServers
         #
         # @return [json] Returns the resource server.
         def resource_server(resource_server_id)
-          raise Auth0::InvalidParameter, 'Must supply a valid resource server id' if resource_server_id.to_s.empty?
+          fail Auth0::InvalidParameter, 'Must supply a valid resource server id' if resource_server_id.to_s.empty?
```
Comment (falvarisqueta) [inline lib/auth0/api/v2/resource_servers.rb:14]: Did you run rubocop? i think fail is a rubocop warning

File: spec/lib/auth0/api/v2/resource_servers_spec.rb [hunk 3]
```diff
@@ -42,11 +41,9 @@
     end
     it 'expect to raise an error when name contains < or > characters' do
       expect { @instance.create_resource_server('test', name: '<') }.to raise_error(
-        'Name must contain at least one character. Does not allow "<" or ">"'
-      )
+        'Name must contain at least one character. Does not allow "<" or ">"')
       expect { @instance.create_resource_server('test', name: '>') }.to raise_error(
-        'Name must contain at least one character. Does not allow "<" or ">"'
-      )
+        'Name must contain at least one character. Does not allow "<" or ">"')
```
Comment (falvarisqueta) [inline spec/lib/auth0/api/v2/resource_servers_spec.rb:46]: i think closing ) in same line is a rubocop warning

**Comment by ignaciojonas** [comment]
Closing, created a new PR #29

**Pull request closed without merging**